src/mds/MDSCacheObject.h

```diff
@@ -253,43 +253,46 @@ class MDSCacheObject {
   // replication (across mds cluster)
  protected:
   unsigned		replica_nonce; // [replica] defined on replica
-  compact_map<mds_rank_t,unsigned>	replica_map;   // [auth] mds -> nonce
+  typedef mempool::mds_co::map<mds_rank_t,unsigned> replica_map_type;
+  alloc_ptr<replica_map_type> replica_map;   // [auth] mds -> nonce
```

Comment:
the main problem of this change is that, operations such as checking if replica_map is empty and iterating replica_map do memory allocation. therefore, most of alloc_ptr will be non-null.

Reply:
Hrm, that's right. I'll try to think of way to resolve this...